### PR TITLE
sci-libs/vtk: fix build against netcdf-4.9.0

### PR DESCRIPTION
**Pull request by waebbl**
Closes: https://bugs.gentoo.org/851594
Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

Please test if this patch works on affected machines.

### CODE REVIEW

File: sci-libs/vtk/files/vtk-9.1.0-avoid-naming-collision-with-netcdf-4.9.0.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+conflicts remove the declaration.
+
+Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
+--- a/ThirdParty/exodusII/vtkexodusII/src/ex_utils.c
++++ b/ThirdParty/exodusII/vtkexodusII/src/ex_utils.c
+@@ -1770,7 +1770,6 @@ void ex__compress_variable(int exoid, int varid, int type)
+         */
+ 
+         /* const int NC_SZIP_EC = 4; */ /* Selects entropy coding method for szip. */
+-        const int NC_SZIP_NN = 32;      /* Selects nearest neighbor coding method for szip. */
```
Comment (thesamesam) [inline sci-libs/vtk/files/vtk-9.1.0-avoid-naming-collision-with-netcdf-4.9.0.patch:18]: Can you use an #ifndef instead?

Comment (waebbl) [reply sci-libs/vtk/files/vtk-9.1.0-avoid-naming-collision-with-netcdf-4.9.0.patch:18]: Good idea, that's less intrusive. I change it.

Comment (waebbl) [reply sci-libs/vtk/files/vtk-9.1.0-avoid-naming-collision-with-netcdf-4.9.0.patch:18]: In this case, I don't need to guard the patch in the build with `has_version`. Do i?

**Comment by waebbl** [comment]
What do we do about the follow up issue from the bug? I couldn't sort this out yet. Can anyone confirm it?

**Comment by thesamesam** [comment]
```
{standard input}:538628: Warning: end of file not at end of a line; newline inserted
{standard input}:538813: Warning: zero assumed for missing expression
{standard input}:538443: Error: invalid operands (*UND* and .gcc_except_table._ZN4vtkm7worklet14marching_cells7executeINS_4cont17CellSetStructuredILi1EEENS3_11ArrayHandleINS_3VecIfLi3EEENS3_23StorageTagUniformPointsEEEdNS3_15StorageTagBasicESB_SB_ffEENS3_17CellSetSingleTypeISB_EERKT_RKT0_RKSt6vectorIT1_SaISL_EERKNS6_ISL_T2_EERNS6_INS7_IT5_Li3EEET3_EERNS6_INS7_IT6_Li3EEET4_EERNS0_7contour11CommonStateE sections) for `-'
{standard input}:538446: Error: invalid operands (*UND* and .gcc_except_table._ZN4vtkm7worklet14marching_cells7executeINS_4cont17CellSetStructuredILi1EEENS3_11ArrayHandleINS_3VecIfLi3EEENS3_23StorageTagUniformPointsEEEdNS3_15StorageTagBasicESB_SB_ffEENS3_17CellSetSingleTypeISB_EERKT_RKT0_RKSt6vectorIT1_SaISL_EERKNS6_ISL_T2_EERNS6_INS7_IT5_Li3EEET3_EERNS6_INS7_IT6_Li3EEET4_EERNS0_7contour11CommonStateE sections) for `-'
aarch64-unknown-linux-gnu-g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
```

I'm 99.99999% sure they ran out of RAM.

**Pull request closed without merging**